src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs

```diff
@@ -3,13 +3,198 @@
 
 namespace Microsoft.DocAsCode.Build.OverwriteDocuments
 {
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    using Markdig.Syntax;
+
+    using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Build.Common;
 
     public class OverwriteDocumentModelCreater
     {
-        public OverwriteDocumentModel Create(MarkdownFragmentModel model)
+        public static OverwriteDocumentModel Create(MarkdownFragmentModel model)
+        {
+            var yamlCocdeBlockMetadata = ConvertYamlCodeBlock(model.YamlCodeBlock, model.YamlCodeBlockSource);
+            var contentsMetadata = ConvertContents(model.Contents);
+            return new OverwriteDocumentModel
+            {
+                Uid = model.Uid,
+                Metadata = yamlCocdeBlockMetadata.Concat(contentsMetadata).GroupBy(p => p.Key)
+                    .ToDictionary(g => g.Key, g => g.Last().Value)
+            };
+        }
+
+        public static Dictionary<string, object> ConvertYamlCodeBlock(string yamlCodeBlock, Block yamlCodeBlockSource)
+        {
+            if (string.IsNullOrEmpty(yamlCodeBlock))
+            {
+                return new Dictionary<string, object>();
+            }
+
+            using (var reader = new StringReader(yamlCodeBlock))
+            {
+                try
+                {
+                    return YamlUtility.Deserialize<Dictionary<string, object>>(reader);
+                }
+                catch (Exception ex)
+                {
+                    throw new MarkdownFragmentsException(
+                        $"Failed when convert yaml code block to `Dictionary<string, object>` with  error: {ex.Message}",
```

Comment:
>Failed when conver [](start = 26, length = 18)

provide a user friendly error message, e.g. "Encountered an invalid YAML code block in line 10: {error detail}"